nb-api/src/main/java/io/nosqlbench/api/engine/metrics/reporters/PromPushReporter.java

```diff
@@ -28,32 +31,82 @@
 import java.net.http.HttpRequest.BodyPublishers;
 import java.net.http.HttpResponse;
 import java.net.http.HttpResponse.BodyHandler;
+import java.nio.file.Path;
 import java.time.Clock;
 import java.time.Duration;
 import java.time.Instant;
 import java.time.ZoneId;
 import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
+import java.util.Optional;
 import java.util.SortedMap;
 import java.util.concurrent.TimeUnit;
 
-public class PromPushReporter extends ScheduledReporter {
+public class PromPushReporter extends ScheduledReporter implements NBConfigurable {
     private static final Logger logger = LogManager.getLogger(PromPushReporter.class);
     private HttpClient client;
     private final URI uri;
+    private String bearerToken;
+    private boolean needsAuth;
 
     public PromPushReporter(
         final String targetUriSpec,
         MetricRegistry registry,
         String name,
         MetricFilter filter,
         TimeUnit rateUnit,
-        TimeUnit durationUnit
+        TimeUnit durationUnit,
+        final String config
     ) {
         super(registry, name, filter, rateUnit, durationUnit);
         uri = URI.create(targetUriSpec);
+        needsAuth = false;
+        ConfigLoader loader = new ConfigLoader();
+        List<Map> configs = loader.load(config, Map.class);
+        NBConfigModel cm = this.getConfigModel();
+        if (configs != null) {
+            logger.info("PromPushReporter process configuration: %s", config);
+            for (Map cmap : configs) {
```

Comment:
This form would represent a number of prom push reporters, so as a revision, this logic should be hoisted out as a wrapper for loading zero or more prom push reporters.

Reply:
I think that there is an opportunity to make all of the metric reporters more configurable essentially as a flattened json. This would work more like the annotators where I grabbed that pattern.